code/game/gamemodes/game_mode.dm

```diff
@@ -475,3 +478,34 @@ proc/get_nt_opposed()
 			text += "<br><b>Objective #[count]</b>: [objective.explanation_text] <span class='boldannounce'>Fail.</span>"
 		count++
 	return text
+
+/datum/game_mode/proc/generate_station_goals()
+	var/list/possible = list()
+	for(var/T in subtypesof(/datum/station_goal))
+		var/datum/station_goal/G = T
+		if(config_tag in initial(G.gamemode_blacklist))
+			continue
+		possible += T
+	var/goal_weights = 0
+	while(possible.len && goal_weights < STATION_GOAL_BUDGET)
+		var/datum/station_goal/picked = pick_n_take(possible)
+		goal_weights += initial(picked.weight)
+		station_goals += new picked
+	
+	if(station_goals.len)	
+		send_station_goals_message()
+	
+/datum/game_mode/proc/send_station_goals_message()

```

Comment:
This is basically the intercept message, except without the intercept and it only mentions the station goals... Not ideal, but I'm not sure about another way to communicate these to the crew.